src/components/ExtractorResponseEditor/ExtractorResponseEditor.tsx

```diff
@@ -422,16 +422,40 @@ class ExtractorResponseEditor extends React.Component<Props, State> {
         document.removeEventListener("Test_SelectWord", this.onTestSelectWord)
     }
 
-    // For end 2 end unit testing.
+    /**
+     * For cypress end to end unit testing.  Event selects phrase in entity labeller
+     * The following will select the word "hello" in the second row (row is 1 based)
+     *   var event = new CustomEvent("Test_SelectWord", { detail: { phrase: "hello", r:ow: 2 }})
+     *   event.initEvent("Test_SelectWord", true, true)}
+     *   element.dispatchEvent(event)
+     * 
+     * Or default to the first row:
+     *   var event = new CustomEvent("Test_SelectWord", { detail: "hello"})
+     *   event.initEvent("Test_SelectWord", true, true)}
+     *   element.dispatchEvent(event)
+     */
     @autobind
     onTestSelectWord(val: any) {
```

Comment:
Is there a way to add type information to get rid of the `any`?  I understand anyone could trigger this event with arbitrary data so we can't guarantee type would need run time checks as you have but after those checks it should narrow the type.

Reply:
onTestSelectWord(val: { detail: { phrase: string, row: number } } | { detail: string }) {
        if (!val.detail) {
            throw new Error("Test_SelectWord expecting detail phrase")
        }

        const phrase: string = (typeof val.detail !== "string") ? val.detail.phrase : val.detail
        const row: number = (typeof val.detail !== "string") ? (val.detail.row - 1) : 0

Reply:
Only thing is then you need to do this:
document.addEventListener("Test_SelectWord", this.onTestSelectWord as any)

Reply:
Why does it have to be `any`? It looks like it could be `EventListenerOrEventListenerObject`

Reply:
Tried that.  Throws a type error